lib/screens/customer_product_detail_page.dart

```diff
@@ -440,14 +443,47 @@ class _CustomerProductDetailPageState extends State<CustomerProductDetailPage> {
           cartItem.stockRequest?.size == _selectedItemSize);
 
       if (productIndex != -1) {
-        customer.cart.items[productIndex].stockRequest.quantity++;
+        StockItem productStock = StockItem();
+        if (_stockRequest.color != null && _stockRequest.size != null) {
+          productStock = _product.stock.items.firstWhere((stockItem) =>
+              stockItem?.color == _selectedItemColor &&
+              stockItem?.size == _selectedItemSize);
+        }
+        if (_stockRequest.color == null && _stockRequest.size != null) {
+          productStock = _product.stock.items
+              .firstWhere((stockItem) => stockItem?.size == _selectedItemSize);
+        }
+        if (_stockRequest.color != null && _stockRequest.size == null) {
+          productStock = _product.stock.items.firstWhere(
+              (stockItem) => stockItem?.color == _selectedItemColor);
+        }
```

Comment:
I think this snippet of code:

```
StockItem productStock = StockItem();
        if (_stockRequest.color != null && _stockRequest.size != null) {
          productStock = _product.stock.items.firstWhere((stockItem) =>
              stockItem?.color == _selectedItemColor &&
              stockItem?.size == _selectedItemSize);
        }
        if (_stockRequest.color == null && _stockRequest.size != null) {
          productStock = _product.stock.items
              .firstWhere((stockItem) => stockItem?.size == _selectedItemSize);
        }
        if (_stockRequest.color != null && _stockRequest.size == null) {
          productStock = _product.stock.items.firstWhere(
              (stockItem) => stockItem?.color == _selectedItemColor);
        }
```

gives the same output as the `_getProductStockItem` method in the `cart_page.dart`. How about making this method public and reusing it in both these places?